File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/AccessTokenHolder.java
package cc.coopersoft.accesstoken.api;


import cc.coopersoft.accesstoken.api.callback.AuthenticationCallback;

import java.util.Optional;

public interface AccessTokenHolder {

  Optional<String> getAccessToken();

  Optional<String> refreshToken();

  void clearToken();

  boolean hasToken();

  void requireToken(Credential credential, AuthenticationCallback callback);


}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/AccessTokenHolder.kt
package cc.coopersoft.accesstoken.api

import cc.coopersoft.accesstoken.api.callback.AuthenticationCallback
import java.util.Optional

interface AccessTokenHolder {
    val accessToken: Optional<String>
    fun refreshToken(): Optional<String>
    fun clearToken()
    fun hasToken(): Boolean
    fun requireToken(credential: Credential?, callback: AuthenticationCallback?)
}